tests/s3/test_async_delete.py

```diff
@@ -0,0 +1,446 @@
+        used_capacity1 = get_cortx_capacity()[-1]
+        self.log.info("CORTX capacity after upload: %s", used_capacity1)
+        self.log.info("Step 2: Delete object 1")
+        resp = self.s3_test_obj.delete_object(self.bucket_name, self.obj_names[0])
+        assert_utils.assert_true(resp[0], resp[1])
+        self.log.info("Step 3: Overwrite object 2")
+        create_random_file(self.upload_paths[2], DEL_CFG["random_obj_size"]["min"],
+                           DEL_CFG["random_obj_size"]["max"])
+        overwrite_resp = self.s3_test_obj.put_object(self.bucket_name, self.obj_names[1],
+                                                     self.upload_paths[2])
+        assert_utils.assert_true(overwrite_resp[0], overwrite_resp[1])
+        overwrite_file_size_2 = os.path.getsize(self.upload_paths[2])
+        self.log.info("Step 4: Verify object deletion")
+        check_get_head_object_version(self.s3_test_obj, self.s3_ver_obj,
+                                      bucket_name=self.bucket_name, object_name=self.obj_names[0],
+                                      get_error_msg=errmsg.NO_SUCH_KEY_ERR,
+                                      head_error_msg=errmsg.NOT_FOUND_ERR)
+        check_list_objects(self.s3_test_obj, bucket_name=self.bucket_name,
+                           expected_objects=[self.obj_names[1]])
+        self.log.info("Step 5: Sleep for GC minimum wait time")
+        time.sleep(DEL_CFG["gc_min_wait_time_min"] * 60)
+        self.log.info("Step 6: Verify space reclaimed")
+        used_capacity2 = get_cortx_capacity()[-1]
+        self.log.info("CORTX capacity after GC: %s", used_capacity2)
+        if overwrite_file_size_1 <= overwrite_file_size_2:
+            assert_utils.assert_true(used_capacity1 <= used_capacity2, "Space not reclaimed.")
+        else:
+            assert_utils.assert_true(used_capacity1 > used_capacity2, "Space not reclaimed.")
+        self.log.info("Step 7: Verify object overwrite")
+        check_get_head_object_version(self.s3_test_obj, self.s3_ver_obj,
+                                      bucket_name=self.bucket_name, object_name=self.obj_names[1],
+                                      etag=overwrite_resp[1]["ETag"])
+        download_and_check(self.s3_test_obj, self.bucket_name, self.obj_names[1],
+                           file_path=self.upload_paths[2], download_path=self.download_path)
+        check_list_objects(self.s3_test_obj, bucket_name=self.bucket_name,
+                           expected_objects=[self.obj_names[1]])
+        self.log.info("ENDED: Test delete and overwrite for simple object having random size")
+
+    @pytest.mark.s3_ops
+    @pytest.mark.tags("TEST-44816")
+    def test_object_deletion_unversioned_bucket_44816(self):
+        """
+        Test object deletion with GC in an unversioned bucket.
+        """
+        self.log.info("STARTED: Test object deletion with GC in an unversioned bucket")
+        self.log.info("Step 1: Upload objects")
```

Comment:
can we write 4 objects here , to see in logs as per tds

Reply:
I have kept it generic in case we need to increase number of object going forward. The sizes and number of objects are specified in config/s3/test_delayed_delete.yaml under fixed_obj_size config option